### DISCUSSION TITLE
Do you think it is consistent behavior for array slicing?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/runtime#59879] [Q&A]
Guess the result of operations.

```CSharp
// 1
new []{1, 2, 3, 4, 5, 6, 7, 8, 9, 0}[10..];

// 2
new []{1, 2, 3, 4, 5, 6, 7, 8, 9, 0}[11..];
```

**Comment by bartonjs** [comment]
1) Empty array.  (Make a copy of the array starting after elements[9] (the 10th item))
2) ArgumentOutOfRangeException (Make a copy of the array starting after elements[10] (the 11th item))

Slicing happens "between" elements.  Slicing 10 means skip the first 10 items.  In the case of a length 11+ collection that would mean "cut it between 10 and 11", but equally valid is "cut it after 10".  Cutting after 10 for a 10-item collection means "return an empty collection".

Slicing 11 means "skip the first 11 items".  In the case of a 10 item collection it throws an exception to represent "I can't find the 11th item".

**Reply by voroninp** [reply]
Then it is surprising that this behavior is not equivalent to `Skip(x).Take(y)` which is mentioned in [docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges#systemrange).
Compare to [F#'s behavior](https://devblogs.microsoft.com/dotnet/announcing-f-5/#consistent-behavior-for-built-in-data-types)
